astropy/coordinates/angles.py

```diff
@@ -664,7 +664,8 @@ def __new__(cls, angle, unit=None, wrap_angle=None, **kwargs):
         self = super().__new__(cls, angle, unit=unit, **kwargs)
         if wrap_angle is None:
             wrap_angle = getattr(angle, 'wrap_angle', self._default_wrap_angle)
-        self.wrap_angle = wrap_angle
+
+        self.wrap_angle = Angle(wrap_angle, copy=False)
```

Comment:
I'm confused. Why is this necessary? the `wrap_angle` setter already does this. Don't the additional tests pass without this?

Reply:
Good catch! I missed the property setter. Correcting it. This will be a no-changelog-needed PR.

Reply:
Good catch! I missed the property setter. This PR will now be a no-changelog-needed since it's just documenting and adding 2 tests.